tests/components/vesync/test_init.py

```diff
@@ -0,0 +1,198 @@
+        assert setups_mock.call_args.args[0] == config_entry
+        assert setups_mock.call_args.args[1] == [Platform.LIGHT]
+        assert setup_mock.call_count == 0
+    assert manager.login.call_count == 1
+    assert hass.data[DOMAIN][VS_MANAGER] == manager
+    assert not hass.data[DOMAIN][VS_SWITCHES]
+    assert not hass.data[DOMAIN][VS_FANS]
+    assert hass.data[DOMAIN][VS_LIGHTS] == [bulb]
+    assert not hass.data[DOMAIN][VS_SENSORS]
+
+
+async def test_async_setup_entry__loads_switches(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    manager: VeSync,
+    switch,
+    dimmable_switch,
+):
+    """Test setup connects to vesync and loads switch platform."""
+    switches = [switch, dimmable_switch]
+    manager.switches = switches
+    manager._dev_list = {
+        "switches": switches,
+    }
+
+    with patch.object(
+        hass.config_entries, "async_forward_entry_setups"
+    ) as setups_mock, patch.object(
+        hass.config_entries, "async_forward_entry_setup"
+    ) as setup_mock:
+        assert await async_setup_entry(hass, config_entry)
+        # Assert platforms loaded
+        await hass.async_block_till_done()
+        assert setups_mock.call_count == 1
+        assert setups_mock.call_args.args[0] == config_entry
+        assert setups_mock.call_args.args[1] == [Platform.SWITCH, Platform.LIGHT]
+        assert setup_mock.call_count == 0
+    assert manager.login.call_count == 1
+    assert hass.data[DOMAIN][VS_MANAGER] == manager
+    assert hass.data[DOMAIN][VS_SWITCHES] == [switch]
+    assert not hass.data[DOMAIN][VS_FANS]
+    assert hass.data[DOMAIN][VS_LIGHTS] == [dimmable_switch]
+    assert not hass.data[DOMAIN][VS_SENSORS]
+
+
+async def test_async_setup_entry__loads_outlets(
+    hass: HomeAssistant, config_entry: ConfigEntry, manager: VeSync, outlet
+):
+    """Test setup connects to vesync and loads switch platform."""
+    outlets = [outlet]
+    manager.outlets = outlets
+    manager._dev_list = {
+        "outlets": outlets,
+    }
+
+    with patch.object(
+        hass.config_entries, "async_forward_entry_setups"
+    ) as setups_mock, patch.object(
+        hass.config_entries, "async_forward_entry_setup"
+    ) as setup_mock:
+        assert await async_setup_entry(hass, config_entry)
+        # Assert platforms loaded
+        await hass.async_block_till_done()
+        assert setups_mock.call_count == 1
+        assert setups_mock.call_args.args[0] == config_entry
+        assert setups_mock.call_args.args[1] == [Platform.SWITCH, Platform.SENSOR]
+        assert setup_mock.call_count == 0
+    assert manager.login.call_count == 1
+    assert hass.data[DOMAIN][VS_MANAGER] == manager
+    assert hass.data[DOMAIN][VS_SWITCHES] == [outlet]
+    assert not hass.data[DOMAIN][VS_FANS]
+    assert not hass.data[DOMAIN][VS_LIGHTS]
+    assert hass.data[DOMAIN][VS_SENSORS] == [outlet]
+
+
+# async def test_unload_entry(hass, config_entry, manager):
```

Comment:
Commented out test.

Reply:
I've been doing working on another vesync PR (https://github.com/home-assistant/core/pull/84025) that has another batch of test improvements.  I'd like to incorporate that test bed before trying to fix this, if possible.  I can remove these test cases, if you would rather not have the extra test methods.

Reply:
I removed the test, but in trying to implement the test, I realized that the implementation is not correct.  It appears that modern integrations store the data  per config entry and this integration does not, yet it's unload acts as though it does.  This fix should go into its own PR.